src/main/java/com/hubspot/jinjava/el/TypeConvertingMapELResolver.java

```diff
@@ -0,0 +1,36 @@
+package com.hubspot.jinjava.el;
+
+import com.hubspot.jinjava.el.ext.CollectionMembershipOperator;
+import java.util.Map;
+import javax.el.ELContext;
+import javax.el.MapELResolver;
+
+public class TypeConvertingMapELResolver extends MapELResolver {
+  private static final TruthyTypeConverter TYPE_CONVERTER = new TruthyTypeConverter();
+
+  public TypeConvertingMapELResolver(boolean readOnly) {
+    super(readOnly);
+  }
+
+  @Override
+  public Object getValue(ELContext context, Object base, Object property) {
+    Object value = super.getValue(context, base, property);
+
+    if (value != null) {
+      return value;
+    }
+
+    if (
+      base instanceof Map &&
+      (Boolean) CollectionMembershipOperator.OP.apply(TYPE_CONVERTER, property, base)
```

Comment:
What does this do separate from the `TYPE_CONVERTER.convert` call later on?

Reply:
It basically makes sure we will be able to convert to the correct type. 
Otherwise it will add an `ELException` to the interpreter errors: `Cannot coerce 'blah' of class java.lang.String to class {enum}`.

Reply:
Ah so the empty map will be covered there.

Reply:
It would be worth just adding a test case with an empty map to cover that then LGTM